### ISSUE TITLE
State becomes stale when action triggered from componentWillMount

### ISSUE THREAD

**Issue by jnfeinstein** [BinaryMuse/fluxxor#93]
I have opted to start a long running action that modifies the state of a store from a `componentWillMount` handler.  In particular, I have launched an ajax request and set the store to reflect that it is loading.  It appears that the `StoreWatchMixin` specifically ignores any state changes in the store until the component is mounted by registering for change events in `componentDidMount` instead of `componentWillMount`.  Consequently the component misses the change event and does not reflect the fact that the store is loading.

The documentation [here](http://fluxxor.com/documentation/store-watch-mixin.html) states that the `StoreWatchMixin` should be used instead of binding to change events in `componentWillMount`.

Is this a bug in Fluxxor or am I misunderstanding Flux/React?

**Comment by BinaryMuse** [comment]
`StoreWatchMixin` was changed in 1c703259485c0d5cf73ad85492e0da0ad172fade to bind to events in `componentDidMount` instead of `componentWillMount` because `willMount` is called on the server, making it easier to leak memory (#88). I failed to update the documentation to match, and I will fix that.

The React docs [suggest firing Ajax requests from `didMount`](http://facebook.github.io/react/docs/component-specs.html#mounting-componentdidmount), though I understand it's nice to be able to get that loading state in the initial state rather than changing it after the fact. If that's something you really want and server rendering is not a concern for you, you might consider just rolling a special version of `StoreWatchMixin` that uses `willMount` instead; it's [a fairly small/simple module](https://github.com/BinaryMuse/fluxxor/blob/3c4b278c4944f5b7b5c24f8d8d8204c40239e2fb/lib/store_watch_mixin.js).
